pkg/importer/http-datasource_test.go

```diff
@@ -313,21 +313,43 @@ var _ = Describe("Http reader", func() {
 		defer ts.Close()
 		ep, err := url.Parse(ts.URL)
 		Expect(err).ToNot(HaveOccurred())
-		r, total, err := createHTTPReader(context.Background(), ep, "", "", "")
+		r, total, _, err := createHTTPReader(context.Background(), ep, "", "", "")
 		Expect(err).ToNot(HaveOccurred())
 		Expect(uint64(25)).To(Equal(total))
 		err = r.Close()
 		Expect(err).ToNot(HaveOccurred())
 	})
 
+	It("should continue even if Content-Length is bogus", func() {
+		redirTs := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			_, _, ok := r.BasicAuth()
+			defer w.WriteHeader(http.StatusOK)
+			w.Header().Add("Content-Length", "intentional gibberish")
+			Expect(ok).To(BeFalse())
+		}))
+		defer redirTs.Close()
+		ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			http.Redirect(w, r, redirTs.URL, http.StatusFound)
+		}))
+		defer ts.Close()
+		ep, err := url.Parse(ts.URL)
+		Expect(err).ToNot(HaveOccurred())
+		r, total, _, err := createHTTPReader(context.Background(), ep, "", "", "")
+		Expect(err).ToNot(HaveOccurred())
+		Expect(uint64(0)).To(Equal(total))
+		err = r.Close()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+
```

Comment:
I am missing a test that checks that if the HEAD fails on something that the return value of the boolean return is true.